### DISCUSSION TITLE
Jetstream, error 419 after session expires

### DISCUSSION THREAD

**Discussion by hugobaret** [laravel/framework#53400] [Q&A]
Hello Laravel community,

I'm working on a project using Laravel, Inertia (Vue 3), and Jetstream, and I've encountered some issues with session expiration.

For testing, I set `SESSION_LIFETIME` to 1 minute. After 1 minute, I’m redirected to the login page, but sometimes I receive a 419 error:

![image](https://github.com/user-attachments/assets/e234b13d-181f-44cd-8134-e01903bd70fa)

If I wait for the session to expire on the login page and then try to log in, I get this error. The same issue occurs when I try to create a team.

Is this behavior expected? Do I need to handle this exception manually, or am I missing something?

I'm using the default Jetstream/Inertia template and have only added features in the Jetstream and Fortify configurations. Redis is set as the `SESSION_DRIVER`, but I experience the same error with `database` as well.

**Comment by noefleury** [comment]
Hi,

This is linked to CSRF token : https://laravel.com/docs/11.x/csrf

See : https://github.com/laravel/framework/blob/cdf07b6678ea64a2a5a9fbe83df9bf4a6586c0a7/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php#L95

And the error 419 is coming from there : https://github.com/laravel/framework/blob/cdf07b6678ea64a2a5a9fbe83df9bf4a6586c0a7/src/Illuminate/Foundation/Exceptions/Handler.php#L639

**Comment by hugobaret** [comment]
Hello again thank you for the answer, I found the solution in the [Error Handling](https://laravel.com/docs/11.x/errors#customizing-the-exception-response) documentation.
